guides-markdown/swaps.md

```diff
@@ -0,0 +1,650 @@
+  publicKey_Bob,
+  timelock
+);
+
+// Derive P2SH address from the script
+const BTC_AddrFromScript = BTC_Swap.getAddressFromRedeemScript(BTC_RedeemScript);
+const BTC_P2SH_Address = BTC_AddrFromScript.toString('testnet');
+
+// Create a watch-only wallet to catch Alice's BTC sweep and get its details
+BTC_WalletInfo = await BTC_WalletClient.createWallet('SwapWithAlice', {watchOnly: true});
+
+// Get the wallet we just created
+const BTC_Wallet = BTC_WalletClient.wallet('SwapWithAlice');
+
+// Import the watch-only address into the wallet's default account
+await BTC_Wallet.importAddress('default', BTC_P2SH_Address);
+
+// Listen for events by "joining" the wallet with it's API token
+await BTC_WalletClient.join('SwapWithAlice', BTC_WalletInfo.token);
+
+// Establish the entire swap-seep procedure and bind it to the watch-only wallet
+BTC_Wallet.bind('confirmed', async (wallet, txDetails) => {
+
+    // Get details from Alice's TX, emitted by the event
+    const fundingTX = BTC_Swap.TX.fromRaw(txDetails.tx, 'hex');
+
+    // Extract the HTLC secret value
+    const revealedSecret = BTC_Swap.extractSecret(
+      fundingTX,
+      BTC_P2SH_Address
+    );
+
+   /*  Create a TX on the BITCOIN CASH chain to sweep Alice's BCH output.
+     *  We assume `BCH_RedeemScript` has already been derived and we have verified
+     *  that Alice has sent the right amount of BCH to that P2SH address.
+     *  Alice's BCH funding TX is stored as `Alices_BCH_FundingTX`, and
+     *  the output that concerns Bob is `Alices_BCH_FundingTX_OutputIndex`.
+     */
+
+    // Derive the input script to redeem the BCH transaction
+    const BCH_SwapScript = BCH_Swap.getSwapInputScript(
+      BCH_RedeemScript,
+      revealedSecret
+    );
+
+    // Build a BCH transaction to spend Alice's output
+    const swapTX = BCH_Swap.getRedeemTX(
+      Bobs_BCH_WalletAddress,
+      feeRate,
+      Alices_BCH_FundingTX,
+      Alices_BCH_FundingTX_OutputIndex,
+      BCH_RedeemScript,
+      BCH_SwapScript,
+      null,
+      Bobs_PrivateKey
+    );
+
+    // Finalize and serialize the TX for broadcast
+    const finalTX = swapTX.toTX();
+    const stringTX = finalTX.toRaw().toString('hex');
+
+    // Broadcast BCH swap-sweep TX, and we're done!
+    const broadcastResult = await BCH_NodeClient.broadcast(stringTX);
+});
+```
+
+Alice will have similar code running on her end. When she broadcasts her BTC sweep
+transaction, Bob's code will automatically catch it, derive the HTLC secret, and
+sweep the BCH transaction.
+
+<iframe width="560" height="315" src="https://www.youtube-nocookie.com/embed/utOAL2Uw_Mk?rel=1" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+
+<a name="next-steps"></a>
+## Next steps
+
+We've built a very simple atomic swaps application! But what have we sacrificed for convenience?
+To improve privacy, we might want to use separate public keys for each blockchain, instead of using
```

Comment:
Could this be done as bullet points?

Reply:
Another next step could be an interface for users to enter unique information (preferred exchange rate?) and also retrieve info that they need to share more conveniently?

Reply:
Right the actual exchange rate! Not even implemented yet :-)

Reply:
Yeah, and the interesting thing with exchange rate, is that's actually dictated by the market, and this would technically be part of the market (thinking broader, how would you even do price discovery if all exchanges are p2p??), so even w/ an exchange rate API suggestion, you might want to just use that as a suggestion, maybe aggregate a few different ones. Doing a p2p exchange might also have a higher premium than centralized exchange because of the privacy benefits. So many possibilities!